### PR TITLE
Apply default docs edit rules to installation docs

### LINKED ISSUE #2208: Installation instructions: Apply default doc edit rules and edit certain headers
In https://github.com/pulumi/pulumi-terraform-bridge/pull/2068, we added the ability to emit `docs/` files to the provider that contain the registry's expected top-level installation doc files (`_index.md` and `installation-configuration.md`), as provided by the upstream `index.md` file. 
Since these installation docs are generated separately from the other registry docs, the [default edit rules](https://github.com/pulumi/pulumi-terraform-bridge/blob/master/pkg/tfgen/edit_rules.go#L26) are not applied via the usual docsgen process. Additionally, we need a few more replaces:
- h/Hashicorp and t/Terraform --> p/Pulumi
- `The following arguments are supported` --> `The following configuration inputs are supported`
- `Argument Reference` --> `Configuration Reference`
- `The foo block contains the following arguments` --> `The foo input has the following nested fields`

Overall, these aim to present a minimal lightweight docs edit pass to make the doc make sense in a Pulumi context.

### PR DESCRIPTION
**Pull request by guineveresaenger**
Adds default doc edit rules plus a few more installation-doc specific edit rules to the plain docs parser functionality.

Fixes #2208

### CODE REVIEW

File: pkg/tfgen/installation_docs.go [hunk 1]
```diff
@@ -23,11 +23,15 @@ func plainDocsParser(docFile *DocFile, g *Generator) ([]byte, error) {
...
-	// - Translation for certain headers such as "Arguments Reference" or "Configuration block"
 	// - Ability to omit irrelevant sections
-	return []byte(contentStr), nil
+
+	// Apply edit rules to transform the doc for Pulumi-ready presentation
+	contentBytes, err := applyEditRules([]byte(contentStr), docFile)
+	if err != nil {
+		return nil, err
+	}
+	return contentBytes, nil
```
Comment (iwahbe) [inline pkg/tfgen/installation_docs.go:34]:
```suggestion
	return applyEditRules([]byte(contentStr), docFile)
```

File: pkg/tfgen/installation_docs.go [hunk 2]
```diff
@@ -91,3 +95,35 @@ func writeInstallationInstructions(goImportBasePath, providerName string) string
...
+		// Reformat certain headers
+		reReplace(`The following arguments are supported`,
+			`The following configuration inputs are supported`),
+		reReplace(`Argument Reference`,
+			`Configuration Reference`),
+		reReplace(`block contains the following arguments`,
+			`input has the following nested fields`),
+	}
+
+	edits = append(edits, installationFileEdits...)
```
Comment (iwahbe) [inline pkg/tfgen/installation_docs.go:120]: I think the shorter version reads better.
```suggestion
	edits := append(defaultEditRules(),
		// Replace all "T/terraform" with "P/pulumi"
		reReplace(`Terraform`, `Pulumi`),
		reReplace(`terraform`, `pulumi`),
		// Replace all "H/hashicorp" strings
		reReplace(`Hashicorp`, `Pulumi`),
		reReplace(`hashicorp`, `pulumi`),
		// Reformat certain headers
		reReplace(`The following arguments are supported`,
			`The following configuration inputs are supported`),
		reReplace(`Argument Reference`,
			`Configuration Reference`),
		reReplace(`block contains the following arguments`,
			`input has the following nested fields`),
	)
```

**Review by iwahbe — COMMENTED** [review]
Looks good to me, but CI is failing.

File: pkg/tfgen/installation_docs.go (hunk 2 above)
Comment (guineveresaenger) [reply pkg/tfgen/installation_docs.go:120]: works for me!

File: pkg/tfgen/installation_docs.go (hunk 1 above)
Comment (guineveresaenger) [reply pkg/tfgen/installation_docs.go:34]: in next PRs I'll be adding more here so I'll need the `contentBytes` for the next transformation 😅

**Review by iwahbe — APPROVED** [review]
LGTM

**Pull request merged**